docs/openstack/barbican_adoption.md

```diff
@@ -0,0 +1,83 @@
+# Barbican adoption
+
+Adopting Barbican means that an existing `OpenStackControlPlane` CR, where Barbican
+is initialy disabled, should be patched to start the service with the configuration
+parameters provided by the source environment.
+
+When the procedure is over, the expectation is to see the `BarbicanAPI`, `BarbicanWorker`,
+`BarbicanKeystoneListener` services up and running.
+The `Keystone endpoints` should also be updated and the same crypto plugin of the source Cloud
+will be available. If the conditions above are met, the adoption is considered concluded.
+
+> **NOTE** Barbican doesn't yet support all of the crypto plugins available in TripleO.
+
+
+## Prerequisites
+
+* Previous Adoption steps completed. Notably, MariaDB and Keystone and [OVN DBs](ovn_adoption.md)
+  should be already adopted.
+
+## Variables
+
+(There are no shell variables necessary currently.)
+
+## Pre-checks
+
+## Procedure - Barbican adoption
+
+* Patch OpenStackControlPlane to deploy Barbican:
+
+  ```
+  oc patch openstackcontrolplane openstack --type=merge --patch '
+  spec:
+    barbican:
+      enabled: true
+      apiOverride:
+        route: {}
+      template:
+        databaseInstance: openstack
+        secret: osp-secret
+        barbicanAPI:
+          replicas: 1
+          override:
+            service:
+              internal:
+                metadata:
+                  annotations:
+                    metallb.universe.tf/address-pool: internalapi
+                    metallb.universe.tf/allow-shared-ip: internalapi
+                    metallb.universe.tf/loadBalancerIPs: 172.17.0.80
+                spec:
+                  type: LoadBalancer
+        barbicanWorker:
+          replicas: 1
+        barbicanKeystoneListener:
+          replicas: 1
+  '
+  ```
+
+## Post-checks
+
+* See that Keystone endpoints are defined and pointing to the podified
+  FQDNs:
+
+  ```bash
+  openstack endpoint list | grep key-manager
+  ```
+
+### Check that Barbican API service is registered in Keystone
+
+```bash
+openstack service list | grep key-manager
+```
+
+```bash
+openstack endpoint list | grep key-manager
+```
+
+### List secrets
+```bash
+openstack secret list
+```
+
```

Comment:
We can do even better in terms of testing by creating a test secret prior to the database migration.  Then, at this stage, we can retrieve this test secret and its payload.  This tests the database migration - but also is a good test of the barbican plugin migration, because the plugin would need to be migrated correctly in order for the secret to be decrypted properly.  We can then delete the test secret.

Reply:
Oh, I see we do that in the test code ..  Feels like a good thing to document here too, eh?

Reply:
Can also do a test for creating a secret (generating a key) - that will test the barbican-workers

Reply:
We will follow up with additional tests in a separate patch. I created a jira story here: https://issues.redhat.com/browse/OSPRH-5317